### DISCUSSION TITLE
Users can enable push protection for themselves

### DISCUSSION THREAD

**Discussion by tuves** [community/community#63294] [Enterprise]
Secret scanning's push protection feature prevents supported secrets from being pushed into repositories, and has to date been enabled at the repository, organization, or enterprise level.

Now, everyone across GitHub can enable push protection for themselves within your individual settings. This ensures your pushes are protected whenever you push to a public repository on GitHub, without relying on that repository to have push protection enabled.

To opt in, go to the ["Code security and analysis"](https://github.com/settings/security_analysis) section of your personal settings. Next to "Push protection for yourself", click Enable.

To learn more, check out the [changelog](https://github.blog/changelog/2023-08-09-users-can-enable-push-protection-for-themselves/).

**Comment by techexpert0119** [comment]
This is fantastic news! The introduction of push protection at the individual level is a significant improvement in securing our code on GitHub. Previously, push protection had to be enabled at the repository, organization, or enterprise level, which meant relying on others to have it enabled. Now, with the option to enable push protection within our individual settings, we have full control over protecting our pushes to public repositories.

Enabling push protection for ourselves ensures that our sensitive secrets are not inadvertently pushed into repositories, reducing the risk of exposure and unauthorized access. It's a great step towards enhancing code security and preventing accidental leaks.

To opt in, I highly recommend everyone to navigate to the "Code security and analysis" section of their personal settings. There, you'll find the option to enable "Push protection for yourself." Just a few clicks, and your pushes to public repositories will be automatically protected.

I appreciate GitHub's commitment to continuously improving security features and providing users with more control over their code. To delve deeper into the specifics of this update, I'll definitely check out the changelog. Thank you for sharing this valuable information!

**Comment by sunlight0102** [comment]
I found the new endpoints for viewing, listing, and updating Dependabot alerts to be very intuitive and easy to use. The API documentation was clear and concise, which made it easy to understand how to interact with the API.

Thank you

**Comment by Binary-Ninja-007** [comment]
Thanks, beauty.

**Comment by xieyuheng** [comment]
We need old feed back:

https://github.com/orgs/community/discussions/66244

**Comment by UncleJooe** [comment]
> Secret scanning's push protection feature prevents supported secrets from being pushed into repositories, and has to date been enabled at the repository, organization, or enterprise level.
> 
> Now, everyone across GitHub can enable push protection for themselves within your individual settings. This ensures your pushes are protected whenever you push to a public repository on GitHub, without relying on that repository to have push protection enabled.
> 
> To opt in, go to the ["Code security and analysis"](https://github.com/settings/security_analysis) section of your personal settings. Next to "Push protection for yourself", click Enable.
> 
> To learn more, check out the [changelog](https://github.blog/changelog/2023-08-09-users-can-enable-push-protection-for-themselves/).

**Comment by devtech5789** [comment]
It sounds like you're discussing a feature or functionality that allows users to enable "push protection" for themselves. Push protection often refers to a security measure aimed at preventing unauthorized or unwanted pushes, such as code changes or updates. This feature can be relevant in various contexts, including software development, access control, or communication platforms.

To provide more detailed information or guidance, I would need additional context or specifics about the system or service in question. If you have any questions or need assistance regarding how to enable push protection in a particular application or service, please provide more details, and I'll be happy to offer guidance.

**Reply by tuves** [reply]
Hi @devtech5789, 

We’ve clarified our stance on using generative AI tools like ChatGPT within our Community via [this announcement](https://github.com/orgs/community/discussions/69006). Please review the guidelines to ensure your post meets them as failure to adhere to those rules can result in action taken by our moderator team.  You can read our updated [Code of Conduct](https://github.com/community/community/blob/baa5cd9020ef3f2bbf86dca1191d9c90f9b0822d/CODE_OF_CONDUCT.md#reasonable-use-of-ai-generated-content) and the [announcement](https://github.com/orgs/community/discussions/69006) for more details. Thank you for helping us maintain an authentic and beneficial space for everyone.
